File: s3GoSDK/sample/GetBucketAcl.go
package main

import (
	"context"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/s3"
)

func GetBucketAcl(arg [3]string) {
	input := &s3.GetBucketAclInput{
		Bucket: aws.String(arg[0]),
	}
	result, err := Client.GetBucketAcl(context.TODO(), input)
	if err != nil {
		fmt.Println("Got an error retrieving ACL for " + arg[0])
		return
	}
	fmt.Println("")
	fmt.Println("Get " + arg[0] + " ACL:")
	fmt.Println("Owner:", *result.Owner.DisplayName)
	fmt.Println("Grants")
	for _, g := range result.Grants {
		// If we add a canned ACL, the name is nil
		if g.Grantee.DisplayName == nil {
			fmt.Println("  Grantee:    EVERYONE")
		} else {
			fmt.Println("  Grantee:   ", *g.Grantee.DisplayName)
		}
		fmt.Println("  Type:      ", string(g.Grantee.Type))
		fmt.Println("  Permission:", string(g.Permission))
		fmt.Println("")
	}
}

File: s3RustSDK/sample/src/bin/get_bucket_acl.rs
use std::process;

mod client;

use s3::model::{BucketCannedAcl};

use structopt::StructOpt;
use tracing_subscriber::fmt::format::FmtSpan;
use tracing_subscriber::fmt::SubscriberBuilder;

#[derive(Debug, StructOpt)]
struct Opt {
    /// The name of the bucket
    #[structopt(short, long)]
    bucket: String,

    /// Whether to display additional information
    #[structopt(short, long)]
    verbose: bool,
}

/// Lists the objects in an Amazon S3 bucket.
/// # Arguments
///
/// * `-n NAME` - The name of the bucket.
/// * `[-d DEFAULT-REGION]` - The region containing the bucket.
///   If not supplied, uses the value of the **AWS_DEFAULT_REGION** environment variable.
///   If the environment variable is not set, defaults to **us-west-2**.
/// * `[-v]` - Whether to display additional information.
#[tokio::main]
async fn main() {
    let Opt {
        bucket,
        verbose,
    } = Opt::from_args();

    if verbose {
        println!("S3 client version: {}", s3::PKG_VERSION);

        SubscriberBuilder::default()
            .with_env_filter("info")
            .with_span_events(FmtSpan::CLOSE)
            .init();
    }

    let acl = BucketCannedAcl::from("log-delivery-write");

    match client::client()
        .put_bucket_acl()
        .bucket(&bucket)
        .acl(acl)
        .send()
        .await{
            Ok(_) => {
                println!("Put bucket acl ");
            }
            Err(e) => {
                println!("Got an error put bucket acl:");
                println!("{}", e);
                process::exit(1);
            }
        }

    match client::client()
        .get_bucket_acl()
        .bucket(&bucket)
        .send()
        .await {
        Ok(resp) => {
            println!("{}'s Acl:",bucket);
            for grant in resp.grants.unwrap_or_default(){
                for grantee in grant.grantee{
                    println!(" Grantee:   {}", grantee.display_name.expect("Grantee have display_name"));
                    for r#type in grantee.r#type{
                        println!(" Type:      {}", r#type.as_str());
                    }
                }
                for permission in grant.permission{
                    println!(" Permission:{}", permission.as_str());
                }
            }
        }
        Err(e) => {
            println!("Got an error retrieving bucket acl:");
            println!("{}", e);
            process::exit(1);
        }
    }
}